### ISSUE TITLE
delete policy  not same as  "kubectl delete"

### ISSUE THREAD

**Issue by yasongxu** [kubernetes/dashboard#5074]
dashboard  use Foreground policy by default when deleting resources ,  it's ok ,  but  it shows 

```
This action is equivalent to:kubectl delete -n default deployment deployment-example
```
![image](https://user-images.githubusercontent.com/12278081/79442061-9ff6c780-800a-11ea-8fa7-2721e20cdd10.png)

kubectl use [Background policy](https://github.com/kubernetes/kubernetes/pull/65908) by default, there is ambiguity here,  I don't know if I understand it correctly 

PTAL;

**Comment by floreks** [comment]
We do not wait for resources to be deleted. Everything is done in background also.

**Comment by yasongxu** [comment]
not the async of front-end page,   but the client-go deleteOptions

code of the dashboard, master branch: 
https://github.com/kubernetes/dashboard/blob/master/src/app/backend/client/verber.go#L180

code of kubectl:
https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/delete/delete.go#L304

am I misunderstanding? 
thanks

**Comment by floreks** [comment]
According to https://github.com/kubernetes/kubernetes/issues/59850 and [documented code](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L438) the result of both deletes should be generally the same. There is no way to force foreground deletion using `kubectl`, but GC should also delete dependant objects once a parent object is deleted. This corresponds to what using foreground deletion does with the only difference being the order of object deletion.

To sum up. 
- Result of UI deletion and `kubectl delete ...` in the end should be the same.
- Difference seems to be only the order of object deletion (foreground will delete parent object last, background should delete it first and then GC should remove orphaned objects) 
- We are using foreground as old K8S versions (1/2y+) had issues with GC not always deleting orphaned objects and foreground deletion worked more reliable. It seems to be fixed now.

**Comment by yasongxu** [comment]
you are right, I agree that foreground is better than background ,  in the end they should be the same.  there have been many discussions in [https://github.com/kubernetes/kubernetes/pull/59851](https://github.com/kubernetes/kubernetes/pull/59851)，and in this commit, kubectl use "background" finally.

sometimes confused, E.g:

when I click "delete" a deployment with many pods on dashboard，sometimes, deployment still exists on the page for a long time, there is no state on the page to indicate whether it has a delete operation, because the page remains unchanged,  then i can click "delete" again , or edit deloyment , it‘s confused... of course, if I know the GC, I can understand it , every client can use its own configuration. 

thanks

**Comment by floreks** [comment]
I think the reason why kubectl uses background delete by default is because CLI waits until object you are trying to delete actually gets deleted. This way it only has to wait for 1 object and controllers will delete dependant objects in background. In our case, we simply send a request to the API and everything happens in background. We do not force user to wait for anything. I guess everything is clear now so I'll close.

/close
